### DISCUSSION TITLE
How to add a button to a widget?

### DISCUSSION THREAD

**Discussion by yablochnyi** [filamentphp/filament#5010] [Help] [answered]
How to add a button to a widget? like on the resource page
 ->headerActions([
                ButtonAction::make('Export')->url(route('export'))->color('primary')->icon('heroicon-o-document-text'),

![1](https://user-images.githubusercontent.com/75489338/202915655-0c27cd28-5958-4dbd-8745-4cd722a57170.JPG)

**Comment by danharrin** [comment, marked as answer]
For table widgets, define `getTableHeaderActions()` and return your actions from there
